### PR TITLE
[NT] Split out `docs` dependencies

### PR DESCRIPTION
**Pull request by timdawborn**
## Description, Motivation and Context

The dependencies in the top-level `package.json` file are not particularly representative of SPOT itself. Many of them, especially the large external libraries (including mobx and react 😱), are only needed for building the docs.

This PR moves all of the docs-only dependencies out to their own `package.json` and `yarn.lock` files. By having the docs dependencies separate, we should have a lot more confidence when upgrading external dependencies that we're not going to break the main application.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,15 +32,27 @@ commands:
       - checkout
       - restore_cache:
           keys:
-            - v3-dependencies-{{ checksum "package.json" }}
+            - v3-dependencies-{{ checksum "yarn.lock" }}
```
Comment (timdawborn) [inline .circleci/config.yml:35]: Unrelated bug fix — the `node_modules` cache should have been keyed by `yarn.lock`, not `package.json`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,15 +32,27 @@ commands:
...
+            - v3-docs-dependencies-{{ checksum "docs/yarn.lock" }}
+            - v3-docs-dependencies
+      - run:
+          name: Install docs dependencies
+          command: yarn install --frozen-lockfile
+          working_directory: docs/
+      - save_cache:
+          paths:
+            - docs/node_modules
+          key: v3-docs-dependencies-{{ checksum "docs/yarn.lock" }}
```
Comment (timdawborn) [inline .circleci/config.yml:55]: Install the `docs` dependencies in addition to the top-level dependencies.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -4,6 +4,16 @@ updates:
...
+    day: monday
+    time: "05:00"
+    timezone: Australia/Sydney
+  open-pull-requests-limit: 99
+- package-ecosystem: npm
+  directory: "/docs"
+  schedule:
+    interval: weekly
+    day: monday
+    time: "05:00"
```
Comment (timdawborn) [inline .github/dependabot.yml:17]: Add dependabot updates for the docs dependencies.

File: .circleci/config.yml [hunk 4]
```diff
@@ -61,31 +77,44 @@ commands:
...
       - store_test_results:
           path: ./test-reports
+      - run:
+          name: Build the documentation.
+          command: |
+            set -uex
+            if [ << parameters.node_version >> -gt 16 ]; then
+              # Without this, we get `Error: error:0308010C:digital envelope routines::unsupported` on Node >= 17.
+              export NODE_OPTIONS="--openssl-legacy-provider"
+            fi
```
Comment (timdawborn) [inline .circleci/config.yml:87]: Build the docs as part of CI.

**Review by njday — APPROVED** [review]
Good idea 👍
